include/robot_interfaces/finger_types.hpp

```diff
@@ -20,3 +20,13 @@ struct FingerTypes : public NJointRobotTypes<3>
 };
 
 }  // namespace robot_interfaces
+
+
+namespace YAML
+{
+template <>
+struct convert<robot_interfaces::FingerTypes::Vector>
+    : robot_interfaces::FingerTypes::yaml_convert_vector
```

Comment:
i think i don't understand what's happening here, could you explain?

Reply:
This creates a convert struct for yaml-cpp as described at the bottom of [its tutorial page](https://github.com/jbeder/yaml-cpp/wiki/Tutorial).
All the implementation is done in `FingerTypes::yaml_convert_vector` but I have to bring it into the `YAML` namespace so I derive from it here to create a kind of alias.  There might be a more elegant solution, I am still lacking some understanding for the more complicated details of how templates work...